### PR TITLE
improvement: Add convenient way to get HTTP Client with sensible timeouts

### PR DESCRIPTION
**Pull request by mikecirioli**
Add convenience methods for getting http.Client with sensible default timeouts, then use the new methods to fix some points in the current code base where we need those timeouts.

- ~~[WIP]~~ unit tests
- ~~[WIP]~~ migrate trouble spots in current codebase to use new methods

#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: pkg/util/http_utils.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+var jxDefaultTransport http.RoundTripper = &http.Transport{
+	DialContext: (&net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+		DualStack: true,
+	}).DialContext,
+	MaxIdleConns:          100,
+	IdleConnTimeout:       90 * time.Second,
+	TLSHandshakeTimeout:   10 * time.Second,
+	ExpectContinueTimeout: 1 * time.Second,
```
Comment (rawlingsj) [inline pkg/util/http_utils.go]: can we make these configurable with env vars maybe, using the values above as defaults?

Comment (mikecirioli) [reply pkg/util/http_utils.go]: sure thing

File: pkg/util/http_utils_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package util
+
+import (
+	"github.com/jenkins-x/jx/src/github.com/stretchr/testify/assert"
```
Comment (rawlingsj) [inline pkg/util/http_utils_test.go]: shouldn't this be "github.com/stretchr/testify/assert"?

Comment (mikecirioli) [reply pkg/util/http_utils_test.go]: possibly, although my IDE provided multiple suggestions and it was not clear which one to choose.  I'll try the one you have suggested.

Comment (rawlingsj) [reply pkg/util/http_utils_test.go]: yeah worth a try

Comment (mikecirioli) [reply pkg/util/http_utils_test.go]: that did the trick, thanks!

Comment (pmuir) [reply pkg/util/http_utils_test.go]: Otherwise you are using a transitive dependency rather than a direct dependency

**Pull request merged**